hazelcast/src/main/java/com/hazelcast/cache/impl/operation/CacheMergeBackupOperation.java

```diff
@@ -0,0 +1,145 @@
+/*
+ * Copyright (c) 2008-2018, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.cache.impl.operation;
+
+import com.hazelcast.cache.CacheEntryView;
+import com.hazelcast.cache.CacheNotExistsException;
+import com.hazelcast.cache.impl.CacheDataSerializerHook;
+import com.hazelcast.cache.impl.ICacheRecordStore;
+import com.hazelcast.cache.impl.ICacheService;
+import com.hazelcast.cache.impl.event.CacheWanEventPublisher;
+import com.hazelcast.cache.impl.record.CacheRecord;
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.Data;
+import com.hazelcast.spi.BackupOperation;
+import com.hazelcast.spi.ObjectNamespace;
+import com.hazelcast.spi.ServiceNamespaceAware;
+import com.hazelcast.spi.impl.AbstractNamedOperation;
+import com.hazelcast.spi.merge.SplitBrainMergePolicy;
+import com.hazelcast.spi.serialization.SerializationService;
+
+import java.io.IOException;
+import java.util.Map;
+
+import static com.hazelcast.cache.impl.CacheEntryViews.createDefaultEntryView;
+import static com.hazelcast.util.MapUtil.createHashMap;
+
+/**
+ * Creates backups for merged {@link CacheRecord} after split-brain healing with a {@link SplitBrainMergePolicy}.
+ *
+ * @since 3.10
+ */
+public class CacheMergeBackupOperation extends AbstractNamedOperation implements BackupOperation, ServiceNamespaceAware {
+
+    private Map<Data, CacheRecord> cacheRecords;
```

Comment:
A collection type can be used to get rid of unneeded hash calculations

Reply:
I think without a convenience wrapper like `MapEntries` this will not be so easy to implement. I would move this to a future enhancement (even if we have to deal with RU then).